### PR TITLE
config/v1 apiserver: add audit.customRules

### PR DESCRIPTION
**Pull request by sttts**
Following enhancements https://github.com/openshift/enhancements/pull/792.

### CODE REVIEW

File: config/v1/types_apiserver.go [hunk 1]
```diff
@@ -76,6 +76,48 @@ const (
...
+	//
+	// If unset, the 'Default' profile is used as the default.
+	//
+	// +kubebuilder:default=Default
+	// +kubebuilder:Enum="",Default,WriteRequestBodies,AllRequestBodies,None
+	Profile AuditProfileType `json:"profile,omitempty"`
+	// customRules specify profiles per group. These profile take precedence over the
+	// top-level profile field if they apply. They are evaluation from top to bottom and
+	// the first one that matches, applies.
+	// +listType=map
```
Comment (p0lyn0mial) [inline config/v1/types_apiserver.go:106]: map?

File: config/v1/0000_10_config-operator_01_apiserver.crd.yaml [hunk 2]
```diff
@@ -47,8 +47,32 @@ spec:
...
+                          - profile
+                        properties:
+                          group:
+                            description: group is a name of group a request user must be member of in order to this profile to apply.
+                            type: string
+                            minLength: 1
+                          profile:
+                            description: "profile specifies the name of the desired audit policy configuration to be deployed to all OpenShift-provided API servers in the cluster. \n The following profiles are provided: - Default: the existing default policy. - WriteRequestBodies: like 'Default', but logs request and response HTTP payloads for write requests (create, update, patch). - AllRequestBodies: like 'WriteRequestBodies', but also logs request and response HTTP payloads for read requests (get, list). - None: no requests are logged at all, not even oauthaccesstokens and oauthauthorizetokens. \n If unset, the 'Default' profile is used as the default."
+                            type: string
+                            enum:
```
Comment (p0lyn0mial) [inline config/v1/0000_10_config-operator_01_apiserver.crd.yaml:67]: we don't have to add the `None` type?

File: config/v1/types_apiserver.go (hunk 1 above)
Comment (sttts) [reply config/v1/types_apiserver.go:106]: map :)

Comment (sttts) [reply config/v1/types_apiserver.go:106]: actually order matters.

Comment (sttts) [reply config/v1/types_apiserver.go:109]: and list type map preserved order 🎉

File: config/v1/0000_10_config-operator_01_apiserver.crd.yaml (hunk 2 above)
Comment (sttts) [reply config/v1/0000_10_config-operator_01_apiserver.crd.yaml:67]: fixed

**Pull request merged**